test/options_test.cc

```diff
@@ -67,8 +67,8 @@ void BM_explicit_iteration_count(benchmark::State& state) {
 
   // Test that the requested iteration count is respected.
   assert(state.max_iterations == 42);
-  size_t actual_iterations = 0;
-  for (auto _ : state) ++actual_iterations;
+  for (auto _ : state) {
```

Comment:
this loop is not doing anything, is it?  Is this something we want in the future?  If so using a `maybe_unused` instead of deleting the variable maybe the way to go.  Or get rid of this code entirely, given that it is not doing anything.

Reply:
I agree, that's how they should have fixed it, and it's how I fixed it locally when I first encountered this.  I am picking the upstream commit intentionally instead of doing that, though, so that when we update the fork, we won't have a conflict.

Reply:
I am under the assumption that the variable named underscore (`_`) is iterating over the `state` collection.  Yes the body of the `for`-loop is empty and thus you may believe it isn't doing anything.  But just looping is doing _something_.  Like it is probably something like:
```cpp
auto it = state.begin();
while( it != state.end() ) it++;
```

So I do believe it wouldn't make sense to get rid of the `for`-loop completely as it might actually be doing something.

Hope that helps some :smiley:

Reply:
but unless there is some really weird behavior attached to the `operator++` then that loop does not do anything.  Maybe the State copy constructor does something really weird as well.  Otherwise we are just assigning each element of `state` to a temporary.  Weird code.